subprojects/docs/src/samples/userguide/tutorial/extraProperties/kotlin/build.gradle.kts

```diff
@@ -21,16 +21,19 @@ task("printTaskProperties") {
 // tag::extraProperties[]
 
 val springVersion by extra("3.1.0.RELEASE")
-val emailNotification by extra("build@master.org")
+val emailNotification by extra { "build@master.org" }
 
-sourceSets.all { extra["purpose"] = null }
+sourceSets.all {
+    this as ExtensionAware
+    extra["purpose"] = null
```

Comment:
@bamboo now that #6715 is merged, I tried removing the cast above. But I get this:
```
  Line 27:         this.extra["purpose"] = "production"
                        ^ Unresolved reference. None of the following candidates is applicable because of receiver type mismatch: 
                            public val ExtensionAware.extra: ExtraPropertiesExtension defined in org.gradle.kotlin.dsl
```
Don't forget to add a `this.` prefix first, to make sure it resolve to the right receiver.

To reproduce:
```
./gradlew integTest:integTest --tests="*.UserGuideSamplesIntegrationTest.userguide_tutorial_extraProperties_*"
```

Even a simple `sourceSets["main"].extra["some"] = "thing"` fails the same.

I didn't get to the bottom of it yet, sharing for the curiosity and in case it rings a bell.

Reply:
After upgrading Kotlin to 1.2.70 this error vanished.